File: PHP/polymorphism.php
<?php 
//Program demonstrates the functionality of polymorphic behaviour in PHP
interface Shape {
    public function calculateArea();
}

class Rectangle implements Shape {
    private $length;
    private $width;

    public function __construct($length, $width) {
        $this->length = $length;
        $this->width = $width;
    }

    public function calculateArea() {
        return $this->length * $this->width;
    }
}

class Circle implements Shape {
    private $radius;

    public function __construct($radius) {
        $this->radius = $radius;
    }

    public function calculateArea() {
        return pi() * $this->radius * $this->radius;
    }
}

// Usage:
$rectangle = new Rectangle(5, 8);
$circle = new Circle(3);

echo "Rectangle Area: " . $rectangle->calculateArea() . "<br>";
echo "Circle Area: " . $circle->calculateArea() . "<br>";

?>

File: Python/Polymorphism.py
# Program demonstrates the functionality of polymorphic behavior in Python

from abc import ABC, abstractmethod
import math

# Define the Shape interface using ABC (Abstract Base Class)
class Shape(ABC):
    @abstractmethod
    def calculate_area(self):
        pass

# Define a class Rectangle that implements the Shape interface
class Rectangle(Shape):
    def __init__(self, length, width):
        self.length = length
        self.width = width

    def calculate_area(self):
        return self.length * self.width

# Define a class Circle that implements the Shape interface
class Circle(Shape):
    def __init__(self, radius):
        self.radius = radius

    def calculate_area(self):
        return math.pi * self.radius * self.radius

# Usage:
rectangle = Rectangle(5, 8)
circle = Circle(3)

print("Rectangle Area:", rectangle.calculate_area())
print("Circle Area:", circle.calculate_area())
